src/main/java/com/atlassian/bitbucket/jenkins/internal/trigger/BitbucketWebhookMultibranchTrigger.java

```diff
@@ -0,0 +1,110 @@
+    public BitbucketWebhookMultibranchTrigger() {
+    }
+
+    @Override
+    public BitbucketWebhookMultibranchTrigger.BitbucketWebhookMultibranchTriggerDescriptor getDescriptor() {
+        return (BitbucketWebhookMultibranchTrigger.BitbucketWebhookMultibranchTriggerDescriptor) super.getDescriptor();
+    }
+
+    @Override
+    public void start(MultiBranchProject<?, ?> project, boolean newInstance) {
+        super.start(project, newInstance);
+        if (!newInstance) {
+            return;
+        }
+        BitbucketWebhookMultibranchTrigger.BitbucketWebhookMultibranchTriggerDescriptor descriptor = getDescriptor();
+        project.getSCMSources().stream().filter(scm -> scm instanceof BitbucketSCMSource)
+                .filter(scm -> ((BitbucketSCMSource) scm).isValid())
+                .forEach(scm -> {
+            boolean isAdded = descriptor.addTrigger(project, (BitbucketSCMSource) scm);
+            ((BitbucketSCMSource) scm).setWebhookRegistered(isAdded);
+        });
+    }
+
+    @Symbol("BitbucketWebhookMultibranchTriggerImpl")
+    @Extension
+    public static class BitbucketWebhookMultibranchTriggerDescriptor extends TriggerDescriptor {
+
+        @Inject
+        private BitbucketPluginConfiguration bitbucketPluginConfiguration;
+        @Inject
+        private RetryingWebhookHandler retryingWebhookHandler;
+
+        @SuppressWarnings("unused")
+        public BitbucketWebhookMultibranchTriggerDescriptor() {
+
+        }
+
+        public BitbucketWebhookMultibranchTriggerDescriptor(RetryingWebhookHandler webhookHandler,
+                                                            BitbucketPluginConfiguration bitbucketPluginConfiguration) {
+            this.retryingWebhookHandler = webhookHandler;
+            this.bitbucketPluginConfiguration = bitbucketPluginConfiguration;
+        }
+
+        @Override
+        public String getDisplayName() {
+            return Messages.BitbucketWebhookMultibranchTrigger_displayname();
+        }
+
+        @Override
+        public boolean isApplicable(Item item) {
+            return item instanceof MultiBranchProject;
+        }
+
+        boolean addTrigger(Item item, BitbucketSCMSource scm) {
+            try {
+                registerWebhook(item, scm.getBitbucketSCMRepository());
+                return true;
+            } catch (Exception ex) {
+                LOGGER.log(Level.SEVERE, "There was a problem while trying to add webhook", ex);
+                throw ex;
+            }
+        }
+
+        private BitbucketServerConfiguration getServer(String serverId) {
+            return bitbucketPluginConfiguration
+                    .getServerById(serverId)
+                    .orElseThrow(() -> new BitbucketClientException(
+                            "Server config not found for input server id" + serverId));
+        }
+
+        private void registerWebhook(Item item, BitbucketSCMRepository repository) {
+            requireNonNull(repository.getServerId());
```

Comment:
```suggestion
```

Reply:
If it's cool, I might leave this as-is, I quite like having the requireNonNull either separate or on variable assignments at the top of the method for readability.

Reply:
Sure, let's introduce a variable then? Honestly, just a nit, so feel free to ignore.
```
private void registerWebhook(Item item, BitbucketSCMRepository repository) {
    String serverId = requireNonNull(repository.getServerId());
    BitbucketServerConfiguration bitbucketServerConfiguration = getServer(serverId);
...
```

Reply:
Sounds good

Reply:
Actually, didn't realise it's merged, so don't worry.